src/main/java/net/minestom/server/terminal/MinestomCompleter.java

```diff
@@ -0,0 +1,39 @@
+package net.minestom.server.terminal;
+
+import net.minestom.server.MinecraftServer;
+import net.minestom.server.command.builder.Command;
+import net.minestom.server.command.builder.suggestion.SuggestionEntry;
+import net.minestom.server.listener.TabCompleteListener;
+import org.jline.reader.Candidate;
+import org.jline.reader.Completer;
+import org.jline.reader.LineReader;
+import org.jline.reader.ParsedLine;
+
+import java.util.List;
+
+public class MinestomCompleter implements Completer {
```

Comment:
You can put this as an inner class in `MinestomTerminal`, there is also no reason to expose it